src/__tests__/providerCollection.spec.ts

```diff
@@ -128,6 +128,137 @@ describe("GET /:provider/collections", () => {
+
+        sandbox.stub(Collections, "getCollections").resolves({
+          count: 0,
+          cursor: null,
+          items: [],
+        });
+
+        const { statusCode, body } = await request(app)
+          .get("/stac/TEST/collections")
+          .query({ q: "NonExistentCollection" });
+
+        expect(statusCode).to.equal(200);
+        expect(body.collections).to.have.lengthOf(0);
+      });
+    });
+
+    describe("given a matching keyword phrase", () => {
+      it("should return collections matching the keyword phrase", async () => {
+        const mockCollections = generateSTACCollections(3);
+        mockCollections[0].title = "Landsat 8 Collection";
+        mockCollections[1].title = "Sentinel-2 Collection";
+        mockCollections[2].title = "MODIS Collection";
+
+        sandbox
+          .stub(Providers, "getProviders")
+          .resolves([null, [{ "provider-id": "TEST", "short-name": "TEST" }]]);
+
+        sandbox.stub(Collections, "getCollections").resolves({
+          count: 1,
+          cursor: null,
+          items: [mockCollections[0]], // Only return the Landsat collection
+        });
+
+        const { statusCode, body } = await request(app)
+          .get("/stac/TEST/collections")
+          .query({ q: '"Landsat 8 Collection"' });
+
+        expect(statusCode).to.equal(200);
+        expect(body.collections).to.have.lengthOf(1);
+        expect(body.collections[0].title).to.equal("Landsat 8 Collection");
+      });
+    });
+
+    describe("given a free text query with a keyword and keyword phrase", () => {
+      it("should return 400 for invalid free text query", async () => {
+        sandbox
+          .stub(Providers, "getProviders")
+          .resolves([null, [{ "provider-id": "TEST", "short-name": "TEST" }]]);
+
+        const { statusCode, body } = await request(app)
+          .get("/stac/TEST/collections")
+          .query({ q: '"Earth Science" Climate' });
+
+        expect(statusCode).to.equal(400);
+        expect(body).to.have.property("errors");
+        expect(body.errors).to.include(
+          "Search query must be either a single keyword or a single phrase enclosed in double quotes."
+        );
+      });
+    });
+
+    describe("given a free text query with unmatched quotes", () => {
+      it("should return 400 for invalid free text query", async () => {
+        sandbox
+          .stub(Providers, "getProviders")
+          .resolves([null, [{ "provider-id": "TEST", "short-name": "TEST" }]]);
+
+        const { statusCode, body } = await request(app)
+          .get("/stac/TEST/collections")
+          .query({ q: '"Earth Science' });
+
+        expect(statusCode).to.equal(400);
+        expect(body).to.have.property("errors");
+        expect(body.errors).to.include(
+          "Search query must be either a single keyword or a single phrase enclosed in double quotes."
+        );
+      });
+    });
+
+    describe("given a free text query with multiple ", () => {
```

Comment:
I think this is supposed to be?
```suggestion
    describe("given a free text query with multiple keyword phrases", () => {
```